### ISSUE TITLE
Borg 1.4.0 remote mount hangs indefinitely

### ISSUE THREAD

**Issue by bulletmark** [borgbackup/borg#8270]
Background:

- Have you checked borgbackup docs, FAQ, and open GitHub issues? Yes
- Is this a BUG / ISSUE report or a QUESTION? Bug
- Version = borg 1.4.0 on both client and server.
- OS = `Linux lt 6.9.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Fri, 28 Jun 2024 04:32:50 +0000 x86_64 GNU/Linux` on both client and server.
- Running with Python 3.12.4 (stock Arch Python).

Arch Linux Borg updated from 1.2.8-2 -> 1.4.0-1 today on my x86_64 server and my x86_64 client. This bug started with 1.4.0. today. I have been using borg for many years. To browse my backup, I usually do:
```sh
$ mkdir tmp
$ borg mount :: tmp
$ cd tmp
$ ls
```
This works normally on my server where the borg backup lives, but hangs indefinitely at the `cd` command on my client (where `BORG_HOST=server` and `BORG_REPO=server:/backup/borg/home`). I find the `borg serve` seems to be running ok on the server while this is happening but the client side `borg mount` is smashing my CPU and my fan comes on.

**Comment by ThomasWaldmann** [comment]
Does a local mount work? Not aware of any change in the "remote" code that could cause this.

borg 1.4-maint branch was forked from 1.2.7 release tag and should be quite similar.

**Comment by bulletmark** [comment]
Yes sorry, this was a problem on my end. I use an aliased [`eza`](https://github.com/eza-community/eza) to list my directories and I recently added the `--git-repos` option (but had forgotten about it). I did the above steps again and waited this time finding that the dir listing appeared after 11 minutes because presumably that option requires it to recurse through the child dirs. I am removing that option!
